lib/pid_file.rb

```diff
@@ -16,7 +16,7 @@ def self.remove(fname)
   def pid
     return nil unless File.file?(@fname)
     data = IO.read(@fname).strip
-    return nil if data.empty?
+    return nil if data.empty? || !/\d+/.match(data)
```

Comment:
Out curiosity, what did the contents of the pid file look like when the was happening?